File: scripts/generateTSadaptorFiles.pl
#!/usr/bin/perl -w
#To use: perl generateTSadaptorFiles.pl Position FULL_PATH_FileList
#       Position: 0-based position in name containing index (File_Cond_AAGGCC_etc.fastq: Position=2)
#	FileList: FULL PATH file list to any unique file in g1Results directory
#	Example: perl generateTSadaptorFiles.pl 2 /proj/seth_lab/smallRNA/PROJ/SAMPLE/*.fastq
# 	J. Baran

$ad_begin='TGGAATTCTCGGGTGCCAAGGAACTCCAGTCAC';
$ad_end = 'ATCTCGTATGCCGTCTTCTGCTTG';
$Position=shift;
foreach $a (@ARGV) {
   $dir =`dirname $a`;
   $base = `basename $a`;
   @parts = split(/\./,$base);

   $ext=$parts[$#parts];
   $base = `basename $a $ext`;
   chomp($dir);
   chomp($ext);
   chomp($base);
   chdir $dir or die;
   @parts = split ("_",$base);
   $AD = join ("",$ad_begin, $parts[$Position],$ad_end);

   $file = $dir . "/" . $base . "adaptor";
   print "$file\n";
   open(FILE, ">$file") or die;
   print FILE "$AD\n";
   close(FILE);

}

File: scripts/generateTSadaptorFiles.py
#!/usr/bin/python2

import sys
import os
import re

adaptor_begin = 'TGGAATTCTCGGGTGCCAAGGAACTCCAGTCAC'
adaptor_end = 'ATCTCGTATGCCGTCTTCTGCTTG'

if sys.argv[1].isdigit():
    adaptor_pos = int(sys.argv[1])
    fastqs = sys.argv[2:]
else:
    fastqs = sys.argv[1:]

for file in fastqs:
    path = os.path.dirname(file) + '/'
    base = os.path.basename(file)
    basename = os.path.splitext(base)[0]
    if 'adaptor_pos' not in locals():
        c = 0
        for bit in basename.split('_'):
            if re.match("^[C,T,G,A]*$", bit):
                    adapt_bit = bit
                    c += 1
        if c != 1:
            print "Could not determine identify adaptor sequence for:", base
            continue
        else:
            adaptor = adaptor_begin + adapt_bit + adaptor_end
            with open('{}{}.adaptor'.format(path, basename), 'w') as f:
                print "Output saved as:", f.name
                f.write('{}\n'.format(adaptor))
    else:
        adapt_bit = basename.split('_')[adaptor_pos]
        if not re.match("^[C,T,G,A]*$", adapt_bit):
            print "Specified adaptor", adapt_bit, "incorrect format for:", base
            continue
        adaptor = adaptor_begin + adapt_bit + adaptor_end
        with open('{}{}.adaptor'.format(path, basename), 'w') as f:
            print "Output saved as:", f.name
            f.write('{}\n'.format(adaptor))

